datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -605,14 +605,12 @@ impl LogicalPlanBuilder {
         let mut join_on: Vec<(Column, Column)> = vec![];
         let mut filters: Option<Expr> = None;
         for (l, r) in &on {
-            if self.plan.schema().field_from_column(l).is_ok()
-                && right.schema().field_from_column(r).is_ok()
-                && can_hash(self.plan.schema().field_from_column(l).unwrap().data_type())
+            if right.schema().field_from_column(r).is_ok()
```

Comment:
This looks like it has changed the semantics of the code. Previously if `field_from_column(l)` was an `Err` then a different code path would have been followed and with this change, there would be an error instead.

Reply:
Ah I see. Really thanks! Fixed!